addons/purchase_stock/models/purchase.py

```diff
@@ -441,9 +441,21 @@ def _prepare_stock_moves(self, picking):
             extra_move_vals['move_dest_ids'] = False  # don't attach
             res.append(extra_move_vals)
         return res
+    
+    def _check_orderpoint_picking_type(self):
+        warehouse_loc = self.order_id.picking_type_id.warehouse_id.view_location_id
+        if self.orderpoint_id.location_id:
```

Comment:
I would prefer using the parent_path to check if the location is below the warehouse
``` python
if not warehouse_loc.parent_path in self.orderpoint.location_id.parent_path:
    raise ...
```

Also a quick test to check the error would be nice. I guess amending an existing test in odoo/addons/purchase_stock/tests/test_reordering_rule.py is good.